app/src/main/java/com/example/vanahel/tasksmanagerapplication/adapter/ViewPagerAdapter.java

```diff
@@ -0,0 +1,39 @@
+package com.example.vanahel.tasksmanagerapplication.adapter;
+
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentPagerAdapter;
+
+import com.example.vanahel.tasksmanagerapplication.tab.AllTabFragment;
+import com.example.vanahel.tasksmanagerapplication.tab.FavoriteTabFragment;
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+
+    int tabCount;
+
+    public ViewPagerAdapter (FragmentManager fragmentManager, int tabCount) {
+        super(fragmentManager);
+        this.tabCount= tabCount;
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        switch (position) {
+            case 0:
+                AllTabFragment allTabFragment = new AllTabFragment();
```

Comment:
ViewPager holds 3 pages in memory by default. It means that if there are more that 3 tabs and user navigates to page #3 and then returns to page #1 the ViewPager will call this method again for the 1st index. It leads to recreation of the first fragment. And if there is any call to DB or to a server it will be re-invoked. Therefore, there is a reason to avoid such recalling, because it might be time or Internet data consuming. Could you please create more smart behavior here that will avoid such recalling?